From 051b1f0bd9b1c50c09ac3453d6bffed6508c3dce Mon Sep 17 00:00:00 2001
From: Michael Natterer <mitch@gimp.org>
Date: Mon, 28 Dec 2015 02:04:46 +0100
Subject: app: fix the curves tool to not warn on layers without alpha

when adding control points to all channels with control-click. It was
looking up curve points based on a -1 alpha value.

(cherry picked from commit e9a38c320d03adda0ae2acd5f6646995a21f4369)
---
 app/tools/gimpcurvestool.c | 25 ++++++++++++++++---------
 1 file changed, 16 insertions(+), 9 deletions(-)

diff --git a/app/tools/gimpcurvestool.c b/app/tools/gimpcurvestool.c
index fa82ba2..acb0c78 100644
--- app/tools/gimpcurvestool.c
+++ app/tools/gimpcurvestool.c
@@ -286,21 +286,26 @@ gimp_curves_tool_button_release (GimpTool              *tool,
     }
   else if (state & gimp_get_toggle_behavior_mask ())
     {
-      gint i;
+      GimpHistogramChannel channel;
 
-      for (i = 0; i < 5; i++)
+      for (channel = GIMP_HISTOGRAM_VALUE;
+           channel <= GIMP_HISTOGRAM_ALPHA;
+           channel++)
         {
-          GimpCurve *curve = config->curve[i];
-          gdouble    value = c_tool->picked_color[i];
+          GimpCurve *curve = config->curve[channel];
+          gdouble    value = c_tool->picked_color[channel];
           gint       closest;
 
-          closest = gimp_curve_get_closest_point (curve, value);
+          if (value != -1)
+            {
+              closest = gimp_curve_get_closest_point (curve, value);
 
-          gimp_curve_view_set_selected (GIMP_CURVE_VIEW (c_tool->graph),
-                                        closest);
+              gimp_curve_view_set_selected (GIMP_CURVE_VIEW (c_tool->graph),
+                                            closest);
 
-          gimp_curve_set_point (curve, closest,
-                                value, gimp_curve_map_value (curve, value));
+              gimp_curve_set_point (curve, closest,
+                                    value, gimp_curve_map_value (curve, value));
+            }
         }
     }
 
@@ -377,6 +382,8 @@ gimp_curves_tool_color_picked (GimpColorTool      *color_tool,
 
   if (gimp_drawable_has_alpha (drawable))
     tool->picked_color[GIMP_HISTOGRAM_ALPHA] = color->a;
+  else
+    tool->picked_color[GIMP_HISTOGRAM_ALPHA] = -1;
 
   tool->picked_color[GIMP_HISTOGRAM_VALUE] = MAX (MAX (color->r, color->g),
                                                   color->b);
-- 
cgit v0.12

From 9c8aeed60df4ec052ea478fa34ccfce4b6fd2a35 Mon Sep 17 00:00:00 2001
From: Massimo Valentini <mvalentini@src.gnome.org>
Date: Mon, 4 Jan 2016 12:55:58 +0100
Subject: Bug 684295: wrong layer group structure in gimp 2.8.2...

... (read PSD file from PS CS3)

use a position of 0 when inserting normal layers
as they're always inserted on top of their group

-1 means inside the group when the active drawable is
a group or above it when it's not. This used to work
before layer groups or in simple cases.

(cherry picked from commit 32469d53b7e28973b3c8488573341c69bef0576b)

 Conflicts:
	plug-ins/file-psd/psd-load.c
---
 plug-ins/file-psd/psd-load.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/plug-ins/file-psd/psd-load.c b/plug-ins/file-psd/psd-load.c
index 35adf82..d789929 100644
--- plug-ins/file-psd/psd-load.c
+++ plug-ins/file-psd/psd-load.c
@@ -1323,7 +1323,7 @@ add_layers (const gint32  image_id,
                                          img_a->columns, img_a->rows,
                                          image_type, 0, GIMP_NORMAL_MODE);
               g_free (lyr_a[lidx]->name);
-              gimp_image_insert_layer (image_id, layer_id, parent_group_id, -1);
+              gimp_image_insert_layer (image_id, layer_id, parent_group_id, 0);
               drawable = gimp_drawable_get (layer_id);
               gimp_drawable_fill (drawable->drawable_id, GIMP_TRANSPARENT_FILL);
               gimp_item_set_visible (drawable->drawable_id, lyr_a[lidx]->layer_flags.visible);
@@ -1360,7 +1360,7 @@ add_layers (const gint32  image_id,
                                          layer_mode);
               IFDBG(3) g_debug ("Layer tattoo: %d", layer_id);
               g_free (lyr_a[lidx]->name);
-              gimp_image_insert_layer (image_id, layer_id, parent_group_id, -1);
+              gimp_image_insert_layer (image_id, layer_id, parent_group_id, 0);
               gimp_layer_set_offsets (layer_id, l_x, l_y);
               gimp_layer_set_lock_alpha  (layer_id, lyr_a[lidx]->layer_flags.trans_prot);
               drawable = gimp_drawable_get (layer_id);
-- 
cgit v0.12

From 242c5152ae96e817042ff869896056d615fd7441 Mon Sep 17 00:00:00 2001
From: Michael Natterer <mitch@gimp.org>
Date: Wed, 6 Jan 2016 05:12:03 +0100
Subject: app: set the name of the open dialog again in file_open_dialog_show()

so we get back the "Open Image as Layers" window title. Seems it was
gone since I broke it in 2008.

(cherry picked from commit 865bb627e01bb9c83ed7f84038a406ef6e2b3db5)
---
 app/actions/file-commands.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/app/actions/file-commands.c b/app/actions/file-commands.c
index 7bb9c20..70b7b4d 100644
--- app/actions/file-commands.c
+++ app/actions/file-commands.c
@@ -527,6 +527,8 @@ file_open_dialog_show (Gimp        *gimp,
       if (uri)
         gtk_file_chooser_set_uri (GTK_FILE_CHOOSER (dialog), uri);
 
+      gtk_window_set_title (GTK_WINDOW (dialog), title);
+
       gimp_file_dialog_set_open_image (GIMP_FILE_DIALOG (dialog),
                                        image, open_as_layers);
 
-- 
cgit v0.12


From d4cebcea8fec7c74b65bd94332d5ac8ce4c303c7 Mon Sep 17 00:00:00 2001
From: Massimo Valentini <mvalentini@src.gnome.org>
Date: Thu, 7 Jan 2016 17:49:52 +0100
Subject: Bug 758760: OpenRaster plug-In could not open image

An oversight while cherry-picking in

621f5e296def8159020201b06f544517624f767d
---
 plug-ins/pygimp/plug-ins/file-openraster.py | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/plug-ins/pygimp/plug-ins/file-openraster.py b/plug-ins/pygimp/plug-ins/file-openraster.py
index 6aa273c..135b9b4 100755
--- plug-ins/pygimp/plug-ins/file-openraster.py
+++ plug-ins/pygimp/plug-ins/file-openraster.py
@@ -271,11 +271,6 @@ def load_ora(filename, raw_filename):
             parent_groups.pop()
             continue
 
-        if not name:
-            # use the filename without extention as name
-            n = os.path.basename(path)
-            name = os.path.splitext(n)[0]
-
         if item.tag == 'stack':
             name, x, y, opac, visible, layer_mode = get_group_layer_attributes(item)
             gimp_layer = gimp.GroupLayer(img)
-- 
cgit v0.12

